caterva2/services/sub.py

```diff
@@ -504,14 +473,21 @@ async def fetch_data(
     Returns
     -------
     StreamingResponse
-        The (slice of) dataset as a NumPy array or a Blosc2 schunk.
+        The (slice of) dataset as a Blosc2 schunk.
     """
 
     slice_ = api_utils.parse_slice(slice_)
     # Download and update the necessary chunks of the schunk in cache
     abspath, dataprep = abspath_and_dataprep(path, slice_, user=user)
     await dataprep()
 
+    # TODO: Also slices covering all data (step 1 or none).
+    whole = slice_ is None or slice_ == () or slice_ == slice(None)
```

Comment:
For the TODO, maybe something along this (not tested)? 
```
if isinstance(slice_, tuple) and len(slice_) > 1: 
    whole *= all(s.step in (None, 1) for s in slice_ if isinstance(s, slice)) 
```

Reply:
Implemented the TODO in commit c8fbecb.